pkg/multicluster/v1/kubeconfigs_snapshot_emitter.sk.go

```diff
@@ -178,6 +197,165 @@ func (c *kubeconfigsEmitter) Snapshots(watchNamespaces []string, opts clients.Wa
 			}
 		}(namespace)
 	}
+	if hasWatchedNamespaces && opts.ExpressionSelector != "" {
+		// watch resources using non-watched namespaces. With these namespaces we
+		// will watch only those that are filted using the label selectors defined
+		// by Expression Selectors
+
+		// first get the renaiming namespaces
+		var k kubernetes.Interface
+		excludeNamespacesFieldDesciptors := ""
+
+		var buffer bytes.Buffer
+		for i, ns := range watchNamespaces {
+			buffer.WriteString("metadata.namespace!=")
+			buffer.WriteString(ns)
+			if i < len(watchNamespaces)-1 {
+				buffer.WriteByte(',')
+			}
+		}
+		excludeNamespacesFieldDesciptors = buffer.String()
+
+		namespacesResources, err := k.CoreV1().Namespaces().List(ctx, metav1.ListOptions{FieldSelector: excludeNamespacesFieldDesciptors})
+		if err != nil {
+			return nil, nil, err
+		}
+		allOtherNamespaces := make([]string, len(namespacesResources.Items))
+		for _, ns := range namespacesResources.Items {
+			allOtherNamespaces = append(allOtherNamespaces, ns.Namespace)
+		}
```

Comment:
nit; performance optimization. initialize size of list up front with `make` and then assign using index in for loop